### PR TITLE
deprecated createNamedQuery and moved functionality to createQuery

### PR DESCRIPTION
**Pull request by Herteby**
createNamedQuery and Collection.createNamedQuery are still there as aliases which show a deprecation warning in the console, and then call createQuery.

### CODE REVIEW

File: lib/query/createQuery.js [hunk 1]
```diff
@@ -1,24 +1,45 @@
...
         throw new Meteor.Error('invalid-query', 'When using createQuery you should only have one main root point that represents the collection name.')
     }
 
-    const entryPointName = _.first(_.keys(data));
+    const entryPointName = _.first(_.keys(body));
 
     const collection = Mongo.Collection.get(entryPointName);
-
     if (!collection) {
+        if(name){ //is a NamedQuery
```
Comment (theodorDiaconu) [inline lib/query/createQuery.js]: space after '(' ... we have to add that linter :D

File: lib/query/extension.js [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+    createQuery(...args) {
+      if (args[0] == 'string') { //NamedQuery
+        const name = args[0];
+        const body = args[1];
+        const params = args[2];
+
+        const query = new NamedQuery(name, this, body, params);
+        NamedQueryStore.add(name, query);
+        return query;
+
```
Comment (theodorDiaconu) [inline lib/query/extension.js]: remove space, leave a blank line before return

**Comment by theodorDiaconu** [comment]
very nice!!

**Comment by zwjcarter** [comment]
There are a couple test files (`/namedQuery/testing/bootstrap/server.js`, `/namedQuery/testing/bootstrap/queries/postListExposure.js`) which use `createNamedQuery`. Not the biggest issue in the world, but I think we should make sure the tests are using the "correct" API. Otherwise it looks nice!

**Comment by Herteby** [comment]
Hehe yeah I know, it looks a bit bad to get warning messages during the test. I left them in though so that the aliases would be tested too.

**Comment by zwjcarter** [comment]
Hmm I hadn't thought of that. Fair enough then!

**Pull request merged**